source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Multipath Events
+category: feature
+authors: Fred Rolland
+feature_name: Multipath Events
+feature_modules: engine,vdsm
+feature_status: Design
+---
+
+# Multipath Events
+
+
+## Overview
+
+This feature adds the ability to display events to the user in
+the engine UI when multipath device path changes its status.
+The supported events are when a path moves to down or up state.
+With these events, the user will be able to detect storage issues faster.
+
+
+## Owner
+
+- Name: [Fred Rolland](https://github.com/rollandf)
+- Email: <frolland@redhat.com>
+
+
+## Detailed Description
+
+
+### How it works
+
+When a host is connected to a DC (Storage Pool), Vdsm will register
+to udev multipath events. For each event, a message will be sent to Engine
+with the details of the event. Engine will create an audit log event. 
+Once the host is disconnected from the DC, Vdsm will unregister from
+the 'udev' events.
+
+
+### New vdsm API
+
+New multipath event type
+
+
+#### Multipath Event
+
+An event with information about the multipath device, path and status.
+
+A mapping with these keys:
+- type: Path Failed/Path Reinstated (DM_ACTION)
+- device (GUID): unique GUID of this device (DM_NAME)
+- valid_paths - number of valid paths (DM_NR_VALID_PATHS)
+- seqnum - event sequence number, may be used to detect missed messages later (DM_SEQNUM)
+
```

Comment:
Let add here a section about the design:

### Vdsm multipath event monitoring

Lets separate the part about reporting events. The design includes 2 components:
1. event listener
2. event notifier(s)

We plan now one event notifier for reporting events to engine. In the future we may add other notifiers, for example for updating lvm filte.